tf/train.py

```diff
@@ -69,44 +56,51 @@ def get_latest_chunks(path, num_chunks, allow_less):
-            [-1, 104, 8]), -1)
-    bit_planes = tf.bitwise.bitwise_and(tf.tile(bit_planes, [1, 1, 1, 8]),
-                                        [128, 64, 32, 16, 8, 4, 2, 1])
-    bit_planes = tf.minimum(1., tf.cast(bit_planes, tf.float32))
-    # Next 5 planes are 1 or 0 to indicate 8x8 of 1 or 0.
-    unit_planes = tf.expand_dims(
-        tf.expand_dims(
-            tf.io.decode_raw(tf.strings.substr(raw, 8272, 5), tf.uint8), -1),
-        -1)
-    unit_planes = tf.tile(unit_planes, [1, 1, 8, 8])
-    # In order to do the conditional for frc we need to make bit unpacked versions.  Note little endian for these fields so the bitwise_and array is reversed.
-    bitsplat_unit_planes = tf.bitwise.bitwise_and(
-        unit_planes, [1, 2, 4, 8, 16, 32, 64, 128])
-    bitsplat_unit_planes = tf.minimum(
-        1., tf.cast(bitsplat_unit_planes, tf.float32))
+    bit_planes = read(7440, 832, tf.uint8)
+    bit_planes = tf.reshape            (bit_planes, [-1, 104, 8])
+    bit_planes = tf.expand_dims        (bit_planes, -1)
+    bit_planes = tf.tile               (bit_planes, [1, 1, 1, 8])
+    bit_planes = tf.bitwise.bitwise_and(bit_planes, [128, 64, 32, 16, 8, 4, 2, 1])
+    bit_planes = tf.cast               (bit_planes, tf.float32)
+    bit_planes = tf.minimum            (bit_planes, 1.)
+
+    # Next 5 bytes are 1 or 0 to indicate 8x8 planes of 1 or 0.
+    unit_planes = read(8272, 5, tf.uint8)
+    unit_planes = tf.expand_dims(unit_planes, -1)
+    unit_planes = tf.expand_dims(unit_planes, -1)
+    unit_planes = tf.tile       (unit_planes, [1, 1, 8, 8])
+    # In order to do the conditional for frc we need to make bit unpacked versions.
+    # Note little endian for these fields so the bitwise_and array is reversed.
+    bitsplat_unit_planes = tf.bitwise.bitwise_and(unit_planes, [1, 2, 4, 8, 16, 32, 64, 128])
+    bitsplat_unit_planes = tf.cast               (bitsplat_unit_planes, tf.float32)
+    bitsplat_unit_planes = tf.minimum            (bitsplat_unit_planes, 1.)
     unit_planes = tf.cast(unit_planes, tf.float32)
-    # rule50 count plane.
-    rule50_plane = tf.expand_dims(
-        tf.expand_dims(
-            tf.io.decode_raw(tf.strings.substr(raw, 8277, 1), tf.uint8), -1),
-        -1)
-    rule50_plane = tf.cast(tf.tile(rule50_plane, [1, 1, 8, 8]), tf.float32)
-    rule50_plane = tf.divide(rule50_plane, 99.)
-    # zero plane and one plane
+
+    # Fifty-move rule count plane.
+    rule50_plane = read(8277, 1, tf.uint8)
+    rule50_plane = tf.expand_dims(rule50_plane, -1)
+    rule50_plane = tf.expand_dims(rule50_plane, -1)
+    rule50_plane = tf.tile       (rule50_plane, [1, 1, 8, 8])
+    rule50_plane = tf.cast       (rule50_plane, tf.float32)
+    rule50_plane = tf.divide     (rule50_plane, 99.)
+
+    # Zero plane and one plane.
     zero_plane = tf.zeros_like(rule50_plane)
     one_plane = tf.ones_like(rule50_plane)
+
     # For FRC unit planes must be replaced with 0 and 2 merged, 1 and 3 merged, two zero planes and then 4.
```

Comment:
didn't touch this section cause it's brand new and it's not as clear how to split it up, but might revisit